src/topp/MzTabExporter.cpp

```diff
@@ -712,7 +662,62 @@ namespace OpenMS
 
       return mztab;
     }
- 
+
+    // generate list of modifications from modified AASequence object. All passed fixed modifications are not reported (PRT and PEP section).
+    // In contrast, modifications are reported in the PSM section (see standard document for details).
+    static MzTabModificationList extractModificationListFromAASequence(const AASequence& aas, const vector<String>& fixed_mods = vector<String>())
+    {
+      MzTabModificationList mod_list;
+      vector<MzTabModification> mods;
+
+      if (aas.isModified())
+      {
+        ModificationsDB* mod_db = ModificationsDB::getInstance();
+
+        if (aas.hasNTerminalModification())
+        {
+          MzTabModification mod;
+          String mod_name = aas.getNTerminalModification();
+          MzTabString unimod_accession = MzTabString(mod_db->getTerminalModification(mod_name, ResidueModification::N_TERM).getUniModAccession());
+          vector<std::pair<Size, MzTabParameter> > pos;
+          pos.push_back(make_pair(0, MzTabParameter()));
+          mod.setModificationIdentifier(unimod_accession);
+          mod.setPositionsAndParameters(pos);
+          mods.push_back(mod);
+        }
+
+        for (Size ai = 0; ai != aas.size(); ++ai)
+        {
+          if (aas.isModified(ai))
+          {
+            MzTabModification mod;
+            String mod_name = aas[ai].getModification();
+            // MzTab standard is to just report Unimod accession.
+            MzTabString unimod_accession = MzTabString(mod_db->getModification(aas[ai].getOneLetterCode(), mod_name, ResidueModification::ANYWHERE).getUniModAccession());
+            vector<std::pair<Size, MzTabParameter> > pos;
+            pos.push_back(make_pair(ai + 1, MzTabParameter()));
+            mod.setPositionsAndParameters(pos);
```

Comment:
Swap this line and the next one, to have the same order in all three cases (N-term, anywhere, C-term). Doesn't make a difference now, but might aid in extending or refactoring later.